Source/UnitTests/GTMSessionFetcherServiceTest.m

```diff
@@ -737,8 +737,11 @@ - (void)testUserAgentFromSessionConfiguration {
 
   XCTestExpectation *expectReceiveResponse = [self expectationWithDescription:@"Received response"];
   NSString *gettysburgPath = [_testServer localPathForFile:kValidFileName];
-  NSString *echoHeadersPath = [gettysburgPath stringByAppendingString:@"?echo-headers=true"];
-  NSURL *validFileURL = [_testServer localURLForFile:echoHeadersPath];
+  NSURL *gettysburgFileURL = [_testServer localURLForFileUsingAppend:gettysburgPath];
+  NSURLComponents *validFileURLComponents = [[NSURLComponents alloc] initWithURL:gettysburgFileURL
+                                                         resolvingAgainstBaseURL:NO];
+  validFileURLComponents.query = @"?echo-headers=true";
+  NSURL *validFileURL = validFileURLComponents.URL;
```

Comment:
The NSURL [query](https://developer.apple.com/documentation/foundation/nsurl/1407543-query) property will be nil if the parameters are appended directly, which will result is a failed test.

Reply:
hm, odd, but ok.